### PR TITLE
Added features for getting stores near the user location

### PR DESCRIPTION
**Pull request by stanco2rhyme**
Added Store locator feature.
Created custom annotation view to dislay the store name and address
added a button to enable the user to go to Apple Map App

### CODE REVIEW

File: BoardBank/BoardBank.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -8,7 +8,6 @@
 
 /* Begin PBXBuildFile section */
 		73E2CEED2279CBA500A7CC0D /* ExtensionRawRepresentable.swift in Sources */ = {isa = PBXBuildFile; fileRef = 73E2CEEC2279CBA500A7CC0D /* ExtensionRawRepresentable.swift */; };
-		7312E64D22792D7A003D863B /* Localizable.strings in Resources */ = {isa = PBXBuildFile; fileRef = 7312E64F22792D7A003D863B /* Localizable.strings */; };
```
Comment (prrenfrew) [inline BoardBank/BoardBank.xcodeproj/project.pbxproj]: Why is this removed? Have you merged the master branch into your feature branch?

File: BoardBank/MonoBank/AddPlayerViewController.swift [hunk 2]
```diff
@@ -5,6 +5,7 @@
 //  Created by Richard Neitzke on 04/01/2017.
 //  Copyright © 2017 Richard Neitzke. All rights reserved.
 //
+// swiftlint:disable all
```
Comment (prrenfrew) [inline BoardBank/MonoBank/AddPlayerViewController.swift]: Don't leave swiftlint disabled for pull requests.

File: BoardBank/MonoBank/GameStores.swift [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+        super.init()
+    }
+    
+    var subtitle: String? {
+        return address
+    }
+    
+    // Annotation right callout accessory opens this mapItem in Maps app
+    func mapItem() -> MKMapItem {
+        let addressDict = [CNPostalAddressStreetKey: subtitle!]
```
Comment (prrenfrew) [inline BoardBank/MonoBank/GameStores.swift]: Don't force unwrap.

File: BoardBank/MonoBank/MapTableViewController.swift [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+//  Copyright © 2019 Richard Neitzke. All rights reserved.
+//
+// swiftlint:disable all
+
+
+import UIKit
+import CoreLocation
+import MapKit
+
+class MapTableViewController: UIViewController {
```
Comment (prrenfrew) [inline BoardBank/MonoBank/MapTableViewController.swift]: What is the purpose of this view controller? Looks like it is doing nothing right now.

File: BoardBank/MonoBank/MapViewController.swift [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+        let request = MKLocalSearch.Request()
+        request.naturalLanguageQuery = "Game Stores"
+        request.region = region
+        
+        let search = MKLocalSearch(request: request)
+        
+        search.start(completionHandler: {(response, error) in
+            
+            if error != nil {
+                print("Error occurred in search: \(error!.localizedDescription)")
```
Comment (prrenfrew) [inline BoardBank/MonoBank/MapViewController.swift]: Don't leave debug print statements in pull requests. Also, don't force unwrap.

File: BoardBank/MonoBank/MapViewController.swift [hunk 6]
```diff
@@ -0,0 +1,131 @@
...
+            } else if response!.mapItems.count == 0 {
```
Comment (prrenfrew) [inline BoardBank/MonoBank/MapViewController.swift]: Use `isEmpty` as opposed to `count == 0`

File: BoardBank/MonoBank/MapViewController.swift [hunk 7]
```diff
@@ -0,0 +1,131 @@
...
+            } else {
+                print("Matches found")
+                
+                for item in response!.mapItems {
+                    guard let coordinate = item.placemark.location?.coordinate else{return}
+                    guard let title =  item.placemark.title else {return}
+                    let name = item.placemark.name
+                    let phoneNumber = item.phoneNumber
+                    //self.matchingItems?.removeAll()
+                    let stores = GameStore(title: name ?? "", locationName: title, address: title, phoneNumber: phoneNumber ?? "", coordinate: coordinate)
```
Comment (prrenfrew) [inline BoardBank/MonoBank/MapViewController.swift]: Looks like it might be better to be able to initialize `GameStore` with a map item.

File: BoardBank/MonoBank/MapViewController.swift [hunk 8]
```diff
@@ -0,0 +1,131 @@
...
+                }
+                print("The Matching stores are : \(self.matchingItems)")
+            }
+        })
+    }
+    
+    // MARK: set the map region
+    func centerMapOnLocation(location: CLLocation) {
+        let coordinateRegion = MKCoordinateRegion(center: location.coordinate,
+                                                  latitudinalMeters: regionRadius, longitudinalMeters: regionRadius)
```
Comment (prrenfrew) [inline BoardBank/MonoBank/MapViewController.swift]: If you are going to spread this over multiple lines, each parameter needs to be put on its own line.

File: BoardBank/MonoBank/MapViewController.swift [hunk 9]
```diff
@@ -0,0 +1,131 @@
...
+        theMapView.setRegion(coordinateRegion, animated: true)
+        let annotation = MKPointAnnotation()
+        annotation.coordinate = location.coordinate
+        self.theMapView.addAnnotation(annotation)
```
Comment (prrenfrew) [inline BoardBank/MonoBank/MapViewController.swift]: Remember to follow the style guide. Style guide states only to use self when needed. It is not needed in this case.

File: BoardBank/MonoBank/MapViewController.swift [hunk 10]
```diff
@@ -0,0 +1,131 @@
...
+        self.theMapView.mapType = .standard
+        theMapView.delegate = self
+        self.queryStores(region: coordinateRegion)
+    }
+    
+    // MARK: Table Button Action Method
+    @IBAction func tableButtonPressed(_ sender: Any) {
```
Comment (prrenfrew) [inline BoardBank/MonoBank/MapViewController.swift]: What is the purpose of this function?

File: BoardBank/MonoBank/MapViewController.swift [hunk 11]
```diff
@@ -0,0 +1,131 @@
...
+    }
+    
+}
+
+// MARK: MKMapViewDelegate Method to configure custom annotation view
+extension MapViewController: MKMapViewDelegate {
+    // 1
```
Comment (prrenfrew) [inline BoardBank/MonoBank/MapViewController.swift]: If you are going to number things, explain what each is doing. This looks like it was copied from elsewhere. Do not copy and paste code. Read and understand the code and come up with your own solution using those ideas.

File: BoardBank/MonoBank/MapViewController.swift [hunk 12]
```diff
@@ -0,0 +1,131 @@
...
+            // 5
+            view = MKMarkerAnnotationView(annotation: annotation, reuseIdentifier: identifier)
+            view.canShowCallout = true
+            view.calloutOffset = CGPoint(x: -5, y: 5)
+            view.rightCalloutAccessoryView = UIButton(type: .detailDisclosure)
+        }
+        return view
+    }
+    
+    func mapView(_ mapView: MKMapView, annotationView view: MKAnnotationView,
```
Comment (prrenfrew) [inline BoardBank/MonoBank/MapViewController.swift]: If spreading over multiple lines, each parameter needs to go on its own line.

File: BoardBank/MonoBank/MapViewController.swift [hunk 13]
```diff
@@ -0,0 +1,131 @@
...
+    }
+}
+// MARK: CLLocationManagerDelegate Method
+extension MapViewController: CLLocationManagerDelegate {
+    
+    
+    func locationManager(_ manager: CLLocationManager, didUpdateLocations locations: [CLLocation]) {
+        
+        guard let currentLocation = locations.last else { return }
+        //        self.location = currentLocation
```
Comment (prrenfrew) [inline BoardBank/MonoBank/MapViewController.swift]: Remove all commented code.

File: BoardBank/MonoBank/MapViewController.swift [hunk 14]
```diff
@@ -0,0 +1,131 @@
...
+        self.manager.stopUpdatingLocation()
+        print("The User current location: \(String(describing: currentLocation)) is this")
+        // Set the Map region based on the user location
+        self.centerMapOnLocation(location: currentLocation)
+        
+    }
+    
+    func locationManager(_ manager: CLLocationManager, didFailWithError error: Error) {
+        print(error.localizedDescription)
```
Comment (prrenfrew) [inline BoardBank/MonoBank/MapViewController.swift]: If there is an error, we need to handle that in some way.

File: BoardBank/MonoBank/MapTableViewController.swift (hunk 4 above)
Comment (stanco2rhyme) [reply BoardBank/MonoBank/MapTableViewController.swift]: I intended to display the game stores in a table view when the user click the Table bar button item

File: BoardBank/MonoBank/GameStores.swift (hunk 3 above)
Comment (stanco2rhyme) [reply BoardBank/MonoBank/GameStores.swift]: Ok

File: BoardBank/BoardBank.xcodeproj/project.pbxproj (hunk 1 above)
Comment (stanco2rhyme) [reply BoardBank/BoardBank.xcodeproj/project.pbxproj]: I thought i mistakenly added it

Comment (stanco2rhyme) [reply BoardBank/BoardBank.xcodeproj/project.pbxproj]: No I have merged yet

File: BoardBank/MonoBank/MapViewController.swift (hunk 5 above)
Comment (stanco2rhyme) [reply BoardBank/MonoBank/MapViewController.swift]: ok

File: BoardBank/MonoBank/MapViewController.swift (hunk 8 above)
Comment (stanco2rhyme) [reply BoardBank/MonoBank/MapViewController.swift]: Ok

File: BoardBank/MonoBank/MapViewController.swift (hunk 9 above)
Comment (stanco2rhyme) [reply BoardBank/MonoBank/MapViewController.swift]: ok

File: BoardBank/MonoBank/MapViewController.swift (hunk 10 above)
Comment (stanco2rhyme) [reply BoardBank/MonoBank/MapViewController.swift]: I intend to display the stores on a table view but maybe i should first finish other required features

File: BoardBank/MonoBank/MapViewController.swift (hunk 11 above)
Comment (stanco2rhyme) [reply BoardBank/MonoBank/MapViewController.swift]: ok

File: BoardBank/MonoBank/MapViewController.swift (hunk 13 above)
Comment (stanco2rhyme) [reply BoardBank/MonoBank/MapViewController.swift]: ok

**Comment by prrenfrew** [comment]
Closing this PR due to not all functionality built and there being merge conflicts.
